CMakeLists.txt

```diff
@@ -31,17 +32,21 @@ else()
 endif()
 
 # =========== unit tests ===================
-if(GOOGLE_TEST_PATH)
-  message(STATUS "GOOGLE_TEST_PATH is set to: ${GOOGLE_TEST_PATH}")
-  find_package(GTest REQUIRED PATHS ${GOOGLE_TEST_PATH} NO_SYSTEM_ENVIRONMENT_PATH)
+if(TESTING_ON)
+  if(GOOGLE_TEST_PATH)
+    message(STATUS "GOOGLE_TEST_PATH is set to: ${GOOGLE_TEST_PATH}")
+    find_package(GTest REQUIRED PATHS ${GOOGLE_TEST_PATH} NO_SYSTEM_ENVIRONMENT_PATH)
+  else()
+    find_package(GTest REQUIRED)
+  endif()
+
+  include(GoogleTest)
+  include(CTest)
+  include(gtest)
 else()
-  find_package(GTest REQUIRED)
+  message(STATUS "Unit tests are disabled")
 endif()
 
-include(GoogleTest)
-include(CTest)
-include(gtest)
-
```

Comment:
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index a75e26e..b51b526 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,16 +31,18 @@ else()
 endif()
 
 # =========== unit tests ===================
-if(GOOGLE_TEST_PATH)
-  message(STATUS "GOOGLE_TEST_PATH is set to: ${GOOGLE_TEST_PATH}")
-  find_package(GTest REQUIRED PATHS ${GOOGLE_TEST_PATH} NO_SYSTEM_ENVIRONMENT_PATH)
-else()
-  find_package(GTest REQUIRED)
-endif()
-
-include(GoogleTest)
 include(CTest)
-include(gtest)
+if(BUILD_TESTING)
+    if(GOOGLE_TEST_PATH)
+      message(STATUS "GOOGLE_TEST_PATH is set to: ${GOOGLE_TEST_PATH}")
+      find_package(GTest REQUIRED PATHS ${GOOGLE_TEST_PATH} NO_SYSTEM_ENVIRONMENT_PATH)
+    else()
+      find_package(GTest REQUIRED)
+    endif()
+
+    include(GoogleTest)
+    include(gtest)
+endif()
 
 # =========== modules ===================
 add_subdirectory(source/modules)

```
There is no need to create a new option, the `CTest` include already provides one ([`BUILD_TESTING`](https://cmake.org/cmake/help/latest/module/CTest.html)).

Reply:
Done